suite-native/labeling/src/components/AccountLabel.tsx

```diff
@@ -1,28 +1,48 @@
 import { useSelector } from 'react-redux';
 
-import { WithLabelingState, selectAccountLabel } from '@suite-common/local-first-storage';
-import { AccountKey, WalletDescriptor } from '@suite-common/wallet-types';
+import {
+    WithLabelingState,
+    selectAccountLabel as selectAccountLabelLocalFirst,
+} from '@suite-common/local-first-storage';
+import { AccountsRootState, selectAccountLabel } from '@suite-common/wallet-core';
+import type { Account } from '@suite-common/wallet-types';
+import { parseDeviceStaticSessionId } from '@suite-common/wallet-utils';
 import { Text } from '@suite-native/atoms';
 
 import { useIsLabelingEnabled } from './useIsLabelingEnabled';
 
-type AccountLabelProps = {
-    walletDescriptor: WalletDescriptor;
-    accountKey: AccountKey;
-    /** @deprecated this shall be removed once Evolu labeling is rolled out and account.accountLabel data migrated */
-    fallbackLabel: string | undefined;
-};
-
-export const AccountLabel = ({
+export const useAccountLabel = ({
     accountKey,
-    walletDescriptor,
-    fallbackLabel,
-}: AccountLabelProps) => {
+    deviceState,
+}: {
+    accountKey: Account['key'] | null;
+    deviceState: Account['deviceState'] | null;
+}) => {
     const isLabelingEnabled = useIsLabelingEnabled();
 
-    const label = useSelector((state: WithLabelingState) =>
-        selectAccountLabel({ state, walletDescriptor, accountKey }),
+    const { walletDescriptor } = deviceState ? parseDeviceStaticSessionId(deviceState) : {};
+
+    const syncedLabel = useSelector((state: WithLabelingState) =>
+        walletDescriptor && accountKey
```

Comment:
Should be the selectors job. It already returns null in some cases, so why not make the parameters optional and in case of undefined values return null from the selector? The same behavior could then be reused everywhere this selector would be used.

Reply:
roger, done